tests/aggregation_regress/tests.py

```diff
@@ -360,6 +361,40 @@ def test_aggrate_annotation(self):
             {'c__max': 3}
         )
 
+    def test_conditional_aggregate_of_subquery(self):
+        """
+        Regression test for #25307: AttributeError: 'WhereNode' object has no
+        attribute 'get_source_expressions'.
+
+        Tests conditional aggregations over subqueries.
+        """
+        # Conditional aggregation on a limited queryset.
```

Comment:
Yes, it's not mentioned on the ticket but any of the 4 cases [here](https://github.com/django/django/blob/master/django/db/models/sql/query.py#L400-L401) would raise the same error.  The test covers the last 3, but I'm not sure how to test the `isinstance(self.group_by, list)` case.  I'll separate out the tests